cdmtaskservice/config_s3.py

```diff
@@ -0,0 +1,80 @@
+"""
+A configuration class for the service for S3 information, including clients.
+"""
+
+from pydantic import BaseModel, Field
+from typing import Optional
+
+from cdmtaskservice.s3.client import S3Client
+
+
+class S3Config(BaseModel):
+    """
+    S3 configuration and client generation.
+    """
+    
+    internal_url: str
+    """ The URL of the S3 instance to use for data storage. """
+    
+    external_url: str
+    """ The URL of the S3 instance accessible to external code or services. """
+    
+    access_key: str
+    """ The S3 access key. """
+    
+    access_secret: str
+    """ The S3 access secret. """
+    
+    error_log_path: str
+    """ A path in S3 where error logs should be stored. """
+    
+    insecure: bool = False
+    """
+    Whether to skip checking the SSL certificate for the S3 instance,
+    leaving the service open to MITM attacks.
+    """
+    
+    verify_external_url: bool = True
+    """ Whether to verify connectivity to the external S3 url at service startup. """
+    
+
+    # Internal singletons
+    _s3_client: Optional[S3Client] = None
+    _s3_external_client: Optional[S3Client] = None
+
+    async def initialize_clients(self):
+        """ Initialize the S3 clients. Calling this method more than once has no effect. """
+        if self._s3_client is None:
+            self._s3_client = await S3Client.create(
```

Comment:
It's okay to initialize clients multiple times, right? In the context with multiple concurrent calls.